sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java

```diff
@@ -240,4 +250,14 @@ private static Value getValue(Resource resource) {
     public String toString() {
         return ToStringBuilder.reflectionToString(this);
     }
+
+    public ResourceBuilder setRole(Optional<String> role) {
```

Comment:
This is a target for a callback.  In the RESERVE, CREATE case, the resourceId generated during reservation must be maintained and using during create.  It's the one somewhat opaque state mutation I really don't like about this PR.  Still thinking how to get rid of it.